### PR TITLE
cinnamon.cinnamon-common: various changes

### LINKED ISSUE #213728: Cinnamon - multiple bugs
### Describe the bug
I have 2 clear bugs for cinnamon that I can't find a way to solve
1- `cinnamon-looking-glass` not working due to missing of `pyinotify`
2- `calendar applet` doesn't show events due to missing of `EDataServer`

both of the issues related to `cinnamon-common`

### Steps To Reproduce
1. `services.xserver.desktopManager.cinnamon.enable = true;` in Configuration.nix
2. run any of the bugged apps

### Notify maintainers
@bobby285271 
@mkg20001 

### Metadata
```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 5.15.90, NixOS, 22.11 (Raccoon), 22.11.2117.22c4a7a4796`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.11.1`
 - channels(root): `"nixos-22.11, nixos-hardware"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

### PR DESCRIPTION
**Pull request by bobby285271**
###### Description of changes

See commit message

I will try to do some of the fix on upstream soon, this file is becoming more and more complex to me...

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by bobby285271** [comment]
@ofborg test cinnamon

**Comment by bobby285271** [comment]
@ofborg test cinnamon

**Comment by bobby285271** [comment]
@ofborg test cinnamon

**Comment by bobby285271** [comment]
@ofborg test cinnamon

Briefly tested the two newly pushed changes

**Comment by bobby285271** [comment]
Okay this reminds me that we are under [`libsoup3` migration](https://people.canonical.com/~ubuntu-archive/transitions/html/libsoup3.html) and this is the start of pulling `libsoup3` stuff in cinnamon package. Sooner or later when cinnamon takes action we need to switch `timezonemap` to its `libsoup3` port and get rid of `libsoup2`.

File: pkgs/desktops/cinnamon/cinnamon-common/default.nix [hunk 1]
```diff
@@ -52,6 +53,23 @@
 , perl
 }:
 
+let
+  pythonEnv = python3.withPackages (pp: with pp; [
+    dbus-python
+    setproctitle
+    pygobject3
+    pycairo
+    python3.pkgs.xapp # The scope prefix is required
```
Comment (SuperSandro2000) [inline pkgs/desktops/cinnamon/cinnamon-common/default.nix:62]: You are really sure on this? Not just xapp?

Comment (bobby285271) [reply pkgs/desktops/cinnamon/cinnamon-common/default.nix:62]: Yes, I tested this on [one of the PR](https://github.com/NixOS/nixpkgs/pull/189824#discussion_r962701929) before, we have both `cinnamon.xapp` and `python3.pkgs.xapp`, they are different package but are both needed here and we have `xapp` as one of the args (for `cinnamon.xapp`) and looks like the arg will have higher priority if we just use `xapp` here. I am unfortunately not able to explain the reason :-(

I wonder if we should just rename `python3.pkgs.xapp` to `python3.pkgs.python-xapp` according to upstream's [meson project name](https://github.com/linuxmint/python3-xapp/blob/master/meson.build#L1).

Comment (bobby285271) [reply pkgs/desktops/cinnamon/cinnamon-common/default.nix]: We can also use `pp.xapp` here.

**Comment by bobby285271** [comment]
@ofborg test cinnamon

**Pull request merged**